### PR TITLE
SwiftCompilerSources: update `Package.swift`

### PR DESCRIPTION
**Pull request by compnerd**
Update the package manager manifest to properly enable C++ interoperability.  This allows us to start trying to build SwiftCompilerSources with SPM again.

### CODE REVIEW

**Review by eeckstein — APPROVED** [review]
lgtm

**Comment by compnerd** [comment]
@swift-ci please test

File: SwiftCompilerSources/Package.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@ import PackageDescription
 
 private extension Target {
   static let defaultSwiftSettings: [SwiftSetting] = [
+    .interoperabilityMode(.Cxx),
     .unsafeFlags([
       "-Xfrontend", "-validate-tbd-against-ir=none",
```
Comment (hyp) [inline SwiftCompilerSources/Package.swift:20]: you can remove these flags, they're set now for interop.

Comment (compnerd) [reply SwiftCompilerSources/Package.swift:20]: Thanks @hyp!

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**